Přidej sraz z 2. 7. 2019 (#44)
PyLadiesCZ/roboprojekt-blog: content/2019-07-02_pokracujeme-se-sitovou-komunikaci.md

Title: Pokračujeme se síťovou komunikací
Date: 2019-07-02 22:28:00
Modified: 2019-07-02 22:28:00
Author: Karolina Surma

> Po dvoutýdenní pauze jsme se opět sešly nad komunikací server-klient. Krok po kroku připisujeme nové funkcionality. Náš server už umí poslat klientovi přiřazeného mu robota a karty, ze kterých si klient vybírá 5 pro zahraní herního kola. Chybí nám odhazovací balíček: pokud serveru dojdou karty na rozdání dalšímu robotovi, měl by vzít všechny doposud odhozené karty a vytvořit z nich nový herní balíček. 

Reviewer: ```suggestion
Po dvoutýdenní pauze jsme se opět sešly nad komunikací server-klient. Krok po kroku připisujeme nové funkcionality. Náš server už umí poslat klientovi jemu přiřazeného robota a karty, ze kterých si klient vybírá 5 pro odehrání herního kola. Chybí nám odhazovací balíček: pokud serveru dojdou karty na rozdání dalšímu robotovi, měl by vzít všechny doposud odhozené karty a vytvořit z nich nový herní balíček. 
```

## Interface

> Změnilo se pozadí - v interface'u dole se nyní vykresluje tolik okének pozadí robotů, kolik na mapě bude hráčů. Ještě řešíme vykreslení avatarů samotných robotů, což určitě budeme mít před dalším srazem hotové.

Reviewer: ```suggestion
Změnilo se pozadí - ve spodní části interface se nyní vykresluje tolik okének pozadí robotů, kolik bude na mapě hráčů. Ještě řešíme vykreslení avatarů samotných robotů, což určitě budeme mít před dalším srazem hotové.
```

> Klient posílá taky serveru vybrané karty s informaci že skončil výběr a zprávu, jestli hraje toto kolo (power down). Server tyto informace umí dekódovat a přeložit na příslušné atributy stavu hry.

Reviewer: ```suggestion
Klient také posílá serveru vybrané karty s informací, že skončil výběr, a zprávu, jestli hraje toto kolo (power down). Server tyto informace umí dekódovat a přeložit na příslušné atributy stavu hry.
```

V dalších krocích naučíme klienta číst zprávu o stavu svého robota ze stavu hry (nyní klientovi posíláme separovaná data, aby si sám vytvořil robota).

## Jména a indikátory

> Přejmenovaly jsme roboty. Doposud nesli jména osob, které je nakreslily v Inkscape'u během prvních týdnů projektu. Nyní mají opravdu kybernetická pojmenování. A zde nás, po přejmenování desítek souborů a spoustě změn v kódu, zastavil Petr. Je totiž rozdíl mezi jménem - tím, které ukazujeme hráči navenek, hezky zformátovaným, s velkými písmeny atd., a indikátorem, s nímž pracujeme uvnitř svého programu a nepropagujeme dál k uživateli. Indikátory není vhodné měnit, jsou zapletené do funkcí programů a každá jejich změna vyžaduje řetězec dalších. Naopak jména jsou čistě jen na nás - herní logika je znát nemusí, patři pouze frontendu. Je to vhodné jak pro vlastní formátovaní jmen, tak pro např. jazykové mutace. A jak dát takové jméno robotovi? Pokud má náš robot poznávací indikátor _franta_, ale přejeme si ho zobrazovat pod jménem _Roy Batty_, můžeme vytvořit soubor `yaml` s název rovným indikátoru, a do těla napsat `name: Roy Batty`. Python v našem frontendu si jméno jednoduše přečte a vykreslí. 

Reviewer: Všimla jsem si, že u anglických termínů nebo názvů programů používáš zvláštní formulaci "Inkscape'u" - to ale v češtině úplně nefunguje, Buď to klasicky skloňuješ, tedy "Inkscapu", nebo já osobně bych se přikláněla k variantě tyto věci v psaném textu neskloňovat, tedy ponechat původní tvar - "v Inkscape".

Author: Ok, přepíšu.

## Alternativní pohled na server

> Petr se před několika srazy zmínil, že on by svůj server navrhl jinak. Abychom si ale užily proces vymyšlení serveru a přidávání mu nových funkcí, hned nám neřekl, jak by tyto otázky řešil on. Jelikož jsme už poměrně daleko v implementaci (konec se blíží!), dnes jsme se dozvěděly, jak bychom to mohly řešit jinak.

Reviewer: ```suggestion
Petr se před několika srazy zmínil, že on by svůj server navrhl jinak. Abychom si ale užily proces vymyšlení serveru a přidávání nových funkcí, hned nám neřekl, jak by tyto otázky řešil on. Jelikož jsme už poměrně daleko v implementaci (konec se blíží!), dnes jsme se dozvěděly, jak bychom to mohly řešit jinak.
```
